### ISSUE TITLE
Application files are not built when `AppConfig.server` is set in `remix.config.js`

### ISSUE THREAD

**Issue by gregory-j-baker** [remix-run/remix#8526]
### Reproduction

<https://stackblitz.com/edit/remix-run-remix-zwfcbj>

1. Run `npm run-script build`

Files of interest: `remix.config.js` and `server.ts`

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.18.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.14.0 - /usr/local/bin/pnpm
  npmPackages:
    @remix-run/css-bundle: ^2.5.0 => 2.5.0 
    @remix-run/dev: ^2.5.0 => 2.5.0 
    @remix-run/express: ^2.5.0 => 2.5.0 
    @remix-run/node: ^2.5.0 => 2.5.0 
    @remix-run/react: ^2.5.0 => 2.5.0 
```

### Used Package Manager

npm

### Expected Behavior

According to [these docs](https://remix.run/docs/en/main/file-conventions/remix-config#server) re `AppConfig.server`:

> **`server`**
> A server entrypoint, relative to the root directory that becomes your server's main module. If specified, Remix will compile this file along with your application into a single file to be deployed to your server. This file can use either a .js or .ts file extension.

### Actual Behavior

When I look in my `build/` directory, I only see a single `server.js` file with no application files, so my application doesn't work.

**Comment by pcattori** [comment]
The client assets are built into `public/build/`, not `build/`

**Comment by gregory-j-baker** [comment]
Thanks @pcattori, that's exactly what I did, and I acknowledge there are numerous examples available for custom servers.

Perhaps the docs can be amended to add another sentence explaining that the `server` field in `remix.config.js` is not meant for a custom `server.?s` file. I showed the docs to a couple of colleagues and neither of them suggested that the `server` config setting was for tools like wrangler, so I feel like my confusion was somewhat justified.
